src/logic/AlarmsManager.ts

```diff
@@ -0,0 +1,50 @@
+import { showMessage, hideMessage } from 'react-native-flash-message';
+
+function AlarmsManager() {
```

Comment:
this is working how I expect it to, but I get this warning:
![image](https://user-images.githubusercontent.com/2033876/81239331-796df000-8ffc-11ea-9401-51b22351dd25.png)

The text is `Cannot update during existing state transition (such as within 'render'). Render methods should be a pure function of props and state.`

I guess that's because I'm calling the `showMessage` in a non react component? I tried with `AlarmsBanner` component but I get the same issue.

@aqilusman45 you may have better React experience, would you happen to know why this might happen?

Reply:
`AlarmsManager.onNewReading(readingValues)`; in `./src/screens/HomeScreen.tsx` is being invoked directly without a `useEffect` hooks which is an equalivent of calling a function in the render method if you were using a class component instead of functional component. This causes a loop behavior and function is invoked during state or props update. To fix this we use component lifecycle method that is `componentDidUpdate` componentor in FC we have have useEffect hooks with `readingValues` in the dependency array, like this, `useEffect(() => {
    AlarmsManager.onNewReading(readingValues);
  }, [readingValues]);` 

Let me know if this fixes the issue. Also HomeScreen component could use some cleaning up, I see unused varaiables, component without child can be self closing.

Reply:
I see. In that case, I will add it elsewhere in the code

Reply:
Awesome, that worked. Will update logic to use alarms banner